File: src/main/java/org/gradle/plugins/site/utils/FileUtils.java
package org.gradle.plugins.site.utils;

import java.io.*;

public final class FileUtils {
    private FileUtils() {}

    public static void createDirectory(File dir) throws IOException {
        if (!dir.exists() && !dir.mkdirs()) {
            throw new IOException("Unable to create directory " + dir);
        }
    }

    public static void copyFile(File source, File target) throws IOException {
        InputStream in = null;
        OutputStream out = null;

        try {
            in = new FileInputStream(source);
            out = new FileOutputStream(target);

            byte[] buf = new byte[1024];
            int len;
            while ((len = in.read(buf)) > 0) {
                out.write(buf, 0, len);
            }
        } finally {
            if (in != null) {
                in.close();
            }
            if (out != null) {
                out.close();
            }
        }
    }
}

File: src/main/kotlin/org/gradle/plugins/site/utils/FileUtils.kt
package org.gradle.plugins.site.utils

import java.io.*

object FileUtils {

    @Throws(IOException::class)
    fun createDirectory(dir: File) {
        if (!dir.exists() && !dir.mkdirs()) {
            throw IOException("Unable to create directory $dir")
        }
    }

    @Throws(IOException::class)
    fun copyFile(source: File, target: File) {
        copyFile(FileInputStream(source), target)
    }

    @Throws(IOException::class)
    fun copyFile(source: InputStream, target: File) {
        var out: OutputStream? = null

        try {
            out = FileOutputStream(target)

            val buf = ByteArray(1024)
            var len: Int = source.read(buf)
            while (len > 0) {
                out.write(buf, 0, len)
                len = source.read(buf)
            }
        } finally {
            source.close()
            out?.close()
        }
    }
}
